index.js

```diff
@@ -19,16 +19,10 @@ util.inherits(CpFileError, NestedError);
 
 CpFileError.prototype.name = 'CpFileError';
 
-module.exports = function (src, dest, opts) {
-	if (!src || !dest) {
-		return Promise.reject(new CpFileError('`src` and `dest` required'));
-	}
-
-	opts = objectAssign({overwrite: true}, opts);
-
-	return new Promise(function startRead(resolve, reject) {
-		var read = fs.createReadStream(src);
+function createReadPromise(src) {
+	var read = fs.createReadStream(src);
```

Comment:
Move this back into the Promise constructor argument (line 26), because `fs.createReadStream` might throw some error (maybe in the future), e.g.
`fs.createReadStream('/dev/null', {flags: 'unicorns'})`